cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_ctrl_packages.go

```diff
@@ -11,3 +11,119 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 package main
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	corev1 "github.com/kubeapps/kubeapps/cmd/kubeapps-apis/gen/core/packages/v1alpha1"
+	datapackagingv1alpha1 "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apiserver/apis/datapackaging/v1alpha1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	log "k8s.io/klog/v2"
+)
+
+// GetAvailablePackageSummaries returns the available packages managed by the 'kapp_controller' plugin
+func (s *Server) GetAvailablePackageSummaries(ctx context.Context, request *corev1.GetAvailablePackageSummariesRequest) (*corev1.GetAvailablePackageSummariesResponse, error) {
+	log.Infof("+kapp-controller GetAvailablePackageSummaries")
+
+	// Retrieve the proper parameters from the request
+	namespace := request.GetContext().GetNamespace()
+	cluster := request.GetContext().GetCluster()
+	pageSize := request.GetPaginationOptions().GetPageSize()
+	pageOffset, err := pageOffsetFromPageToken(request.GetPaginationOptions().GetPageToken())
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "unable to intepret page token %q: %v", request.GetPaginationOptions().GetPageToken(), err)
+	}
+	// Assume the default cluster if none is specified
+	if cluster == "" {
+		cluster = s.globalPackagingCluster
```

Comment:
Same thought as earlier... why do we not just require cluster in the request now?

Reply:
I guess

Reply:
No no, all good. I'd mistakenly thought that we'd updated already elsewhere, but what it was was that we updated the flux plugin to accept and use it when set (since the dashboard always sends it). Better to have it consistent with the other plugins. Sorry for the noise.